tap_tiktok/streams.py

```diff
@@ -255,6 +255,16 @@ def get_url_params(
             start_date = datetime.datetime.strptime(next_page_token["start_date"], DATE_FORMAT)
         else:
             start_date = self.get_starting_timestamp(context)
+
+        lookback_window = self.config["lookback"]
+        if lookback_window > 0:
+            # if lookback is configured, we want to refetch data for the entire lookback window
+            # (or as far back as the configured start date, whichever is the most recent date)
+            start_date = max(
```

Comment:
would this override the `start_date` from the pagination if `lookback` is set? i.e. Should this happen where we fetch the initial `start_date` on line 257? And then just be the bookmarked value minus the `lookback`

Reply:
Good catch! I have moved the lookback handling into the `else` block where the initial or bookmarked start date is loaded, so it will no longer override start date from pagination.